agent/rpc_to.go

```diff
@@ -96,3 +139,35 @@ func (a *orbAgent) sendAgentPoliciesReq() error {
 
 	return nil
 }
+
+func (a orbAgent) retryAgentPolicyResponse() {
```

Comment:
it's a shame (and a maintenance burden) to reproduce all of this complex logic - is there any way to make this a single function used by both group and policy (and maybe others in the future?). maybe we can flag it as tech debt if nothing else.

Reply:
yeah, I am assembling something to demonstrate, I did this on the previous project that fixed most of the issues, by adding names to all routines, and pass along context, and instead of channels for tracking the info with routine and cancel functions. 
and also to divide this code to more adherent to single responsibility principle... I will open a ticket with this.